okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.kt

```diff
@@ -129,16 +129,24 @@ class HttpLoggingInterceptor @JvmOverloads constructor(
 
   fun redactHeader(name: String) {
     val newHeadersToRedact = TreeSet(String.CASE_INSENSITIVE_ORDER)
-    newHeadersToRedact.addAll(headersToRedact)
-    newHeadersToRedact.add(name)
+    newHeadersToRedact += headersToRedact
+    newHeadersToRedact += name
     headersToRedact = newHeadersToRedact
   }
 
+  @Deprecated(
+      message = "Moved to var. Replace setLevel(...) with level(...) to fix Java",
+      replaceWith = ReplaceWith(expression = "run { this.level = level }"),
```

Comment:
This looks wrong, I would have thought run should be apply here.

https://proandroiddev.com/the-difference-between-kotlins-functions-let-apply-with-run-and-else-ca51a4c696b8

run should be returning the level as the result instead of the interceptor.

Reply:
Yeah, I see your point. `apply { this.level = level; this }` gives a way for the user to do 100% compatible replacement for a while. Once we remove the deprecated `setLevel()`, they'll end up with changing the code into `this.level = level` later. :eyes:

Reply:
you shouldn't even need the "; this".  apply should execute the block but still return the interceptor.

Reply:
ah yeah, you're right! `; this` shouldn't be there for `apply(...)`. Thanks!
`public inline fun <T> T.apply(block: T.() -> Unit): T`
`public inline fun <T, R> T.run(block: T.() -> R): R`

After doing this kind of stuff in #5144 , it feels to me that `apply { this.level = level }` would be nice here? 🤔

Reply:
+1 to apply

Reply:
Fixed.